preview_generator/preview/generic_preview.py

```diff
@@ -32,6 +34,27 @@ def check_dependencies(cls) -> None:
         if a dependency is missing.
         """
 
+    @classmethod
+    def get_mimetypes_mapping(cls) -> typing.List[MimetypeMapping]:
+        """
+        Get specific mimetypes mappings (mimetype/file_extension) related to
+        builder, this allow to update mimetypes mapping use by preview_generator
+        to help preview_generator to determine more correctly type or file_extension
+        :return:
+        """
+        return []
+
+    @classmethod
+    def update_mimetypes_mapping(cls) -> None:
+        """
+        Update mimetypes mapping with file extension in preview_generator
+        mimetypes_storage
+        """
+        for mimetypes_mapping in cls.get_mimetypes_mapping():
+            mimetypes_storage.add_type(  # type: ignore
+                type=mimetypes_mapping.mimetype, ext=mimetypes_mapping.file_extension, strict=False
+            )
```

Comment:
`mimetypes_storage` should be given by parameter (more flexible for developpers want to use their own `mimetypes_storage` objet)

Reply:
Don't see any easy way to do this. mimetypes_storage is needed at many place.

Reply:
will be solved later https://github.com/algoo/preview-generator/issues/159